src/test/rustdoc/for-lifetime.rs

```diff
@@ -0,0 +1,14 @@
+// ignore-tidy-linelength
+
+#![crate_name = "foo"]
+#![crate_type = "lib"]
+
+pub struct Foo {
+    pub some_func: for<'a> fn(val: &'a i32) -> i32,
+    pub some_trait: dyn for<'a> Trait<'a>,
+}
+
+// @has foo/index.html '//span[@id="structfield.some_func"]' "some_func: fn<'a>(val: &'a i32) -> i32"
+// @has foo/index.html '//span[@id="structfield.some_trait"]' "some_trait: dyn Trait<'a>"
```

Comment:
These tests are failing :/

I wish `@has` checks gave me something better than:

```
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------
11: @has check failed
	`XPATH PATTERN` did not match
	// @has foo/index.html '//span[@id="structfield.some_func"]' "some_func: fn<'a>(val: &'a i32) -> i32"
12: @has check failed
	`XPATH PATTERN` did not match
	// @has foo/index.html '//span[@id="structfield.some_trait"]' "some_trait: dyn Trait<'a>"

Encountered 2 errors

------------------------------------------

info: generating a diff against nightly rustdoc
warning: no pager configured, falling back to `diff --color`
help: try configuring a git pager (e.g. `delta`) with `git config --global core.pager delta`
thread '[rustdoc] rustdoc/for-lifetime.rs' panicked at 'assertion failed: output.status.success() || output.status.code() == Some(1)', src/tools/compiletest/src/runtest.rs:2469:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    [rustdoc] rustdoc/for-lifetime.rs
```

@jyn514 Didn't you make a PR at some point to add nice diffs? It seems like it doesn't work on my machine...

Reply:
@camelid did you try following the advice? ;)

```
info: generating a diff against nightly rustdoc
warning: no pager configured, falling back to `diff --color`
help: try configuring a git pager (e.g. `delta`) with `git config --global core.pager delta`
```

Reply:
What's a "diff pager"?

Reply:
I tried `git config --global core.pager less`, but it still doesn't work. It doesn't use the pager and just outputs a huge unstyled diff.

Reply:
Oh wait actually this is probably https://github.com/rust-lang/rust/pull/79370

Reply:
Try `brew install tidy` (or however you prefer to install programs)

Reply:
I already have tidy installed :)

Reply:
Also, I'm not getting that error.

Reply:
@camelid I copy-pasted that output from the error you posted in https://github.com/rust-lang/rust/pull/79991#discussion_r541808910. Are you getting a different error now?

Reply:
Also, it might help to rebase over https://github.com/rust-lang/rust/pull/79370.

Reply:
I was referring to the error in #79370 which is about `tidy`, not the error I posted which is about a pager.

Reply:
@camelid I added more tools for debugging this in https://github.com/rust-lang/rust/pull/79370, can you rebase over that and see what the new output is when you run the test?

Reply:
I rebased over `upstream/master` but I'm getting this error:

```
stdout:
------------------------------------------

------------------------------------------
stderr:
------------------------------------------
11: @has check failed
	`XPATH PATTERN` did not match
	// @has foo/index.html '//span[@id="structfield.some_func"]' "some_func: fn<'a>(val: &'a i32) -> i32"
12: @has check failed
	`XPATH PATTERN` did not match
	// @has foo/index.html '//span[@id="structfield.some_trait"]' "some_trait: dyn Trait<'a>"

Encountered 2 errors

------------------------------------------

info: generating a diff against nightly rustdoc
thread '[rustdoc] rustdoc/for-lifetime.rs' panicked at 'assertion failed: status.success() || status.code() == Some(1)', src/tools/compiletest/src/runtest.rs:2420:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    [rustdoc] rustdoc/for-lifetime.rs
```

Then I edited compiletest to show what the actual status code is:

```
thread '[rustdoc] rustdoc/for-lifetime.rs' panicked at 'HTML `tidy` returned unexpected status code Some(2)', src/tools/compiletest/src/runtest.rs:2420:21
```

Then I removed the assert and changed `diff --color` to `diff` since my version of diff doesn't seem to have a `--color` option, and it dumps a massive diff. What should I do?

Reply:
> Then I removed the assert and changed diff --color to diff since my version of diff doesn't seem to have a --color option,

This is a bug in compiletest, can you open an issue?

>  it dumps a massive diff

This is kind of just a limitation of how I wrote it :/ if you run `rustup default nightly` (or `rustup override set nightly`, which doesn't affect global state) it should at least have a *smaller* diff. It will never be exactly what you want to change though, that would require comparing to `git merge-base HEAD origin/master` which I don't have the infrastructure set up for, and I'm not quite sure how I'd add it.

Reply:
Opened #80037.

Reply:
It's also not paging the output, even though I set my pager to `less`.

Reply:
I managed to eventually generate a diff using this command:

```console
$ git diff --no-index --word-diff -- build/x86_64-apple-darwin/test/rustdoc/for-lifetime.nightly build/x86_64-apple-darwin/test/rustdoc/for-lifetime
```

```diff
<div class="docblock type-decl hidden-by-usual-hider"><pre class="rust struct">pub struct Foo {
    pub some_func: {+for<'a>+} <a class="primitive" [-href="https://doc.rust-lang.org/nightly/std/primitive.fn.html">fn</a>'a(val:-]{+href="https://doc.rust-lang.org/nightly/std/primitive.fn.html">fn</a>(val:+} &amp;'a <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.i32.html">i32</a>) -&gt; <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.i32.html">i32</a>,
    pub some_trait: dyn <a class="trait" href="../foo/trait.Trait.html" title="trait foo::Trait">Trait</a>&lt;'a&gt;,
}</pre></div><h2 id="fields" class="fields small-section-header">
                       Fields<a href="#fields" class="anchor"></a></h2><span id="structfield.some_func" class="structfield small-section-header"><a href="#structfield.some_func" class="anchor field"></a><code>some_func: {+for<'a>+} <a class="primitive" [-href="https://doc.rust-lang.org/nightly/std/primitive.fn.html">fn</a>'a(val:-]{+href="https://doc.rust-lang.org/nightly/std/primitive.fn.html">fn</a>(val:+} &amp;'a <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.i32.html">i32</a>) -&gt; <a class="primitive" href="https://doc.rust-lang.org/nightly/std/primitive.i32.html">i32</a></code></span>
```

Reply:
Fixed! https://github.com/rust-lang/rust/pull/79991#issuecomment-762553369